pubsub/subscriptions/main.go

```diff
@@ -163,3 +165,56 @@ func createTopicIfNotExists(c *pubsub.Client) *pubsub.Topic {
 	}
 	return t
 }
+
+func getPolicy(c *pubsub.Client, subName string) *iam.Policy {
+	ctx := context.Background()
+
+	// [START pubsub_get_subscription_policy]
+	policy, err := c.Subscription(subName).IAM().Policy(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, role := range policy.Roles() {
+		log.Printf("%q: %q", role, policy.Members(role))
+	}
+	// [END pubsub_get_subscription_policy]
+	return policy
+}
+
+func addUsers(c *pubsub.Client, subName string) {
+	ctx := context.Background()
+
+	// [START pubsub_set_subscription_policy]
+	sub := c.Subscription(subName)
+	policy, err := sub.IAM().Policy(ctx)
+	if err != nil {
+		log.Fatalf("GetPolicy: %v", err)
+	}
+	// Other valid prefixes are "serviceAccount:", "user:"
+	// See the documentation for more values.
+	policy.Add(iam.AllUsers, iam.Viewer)
+	policy.Add("group:cloud-logs@google.com", iam.Editor)
+	if err := sub.IAM().SetPolicy(ctx, policy); err != nil {
```

Comment:
```
+       // NOTE: It may be necessary to retry this operation if IAM policies are
+       // being modified concurrently. SetPolicy will return an error if the policy
+       // was modified since it was retrieved.
```